go/coinstacks/osmosis/tx.go

```diff
@@ -29,13 +29,64 @@ func ParseMessages(msgs []sdk.Msg, events cosmos.EventsByMsgIndex) []cosmos.Mess
 			}
 			messages = append(messages, message)
 		case *gammtypes.MsgJoinPool:
-			message := cosmos.Message{
-				Addresses: []string{v.Sender},
-				From:      v.Sender,
-				Type:      v.Type(),
-				Value:     cosmos.CoinToValue(&v.TokenInMaxs[0]),
+			tokenInValues := cosmos.PoolEventAmountToValues(events["0"]["pool_joined"]["tokens_in"])
+			msgs := []cosmos.Message{
+				// token in 0
+				{
+					Addresses: []string{v.Sender},
+					Origin:    v.Sender,
+					From:      v.Sender,
```

Comment:
ideally we would want the `To` address, however there is an issue with event parsing in which keys with the same name overwrite each other... I wouldn't think this would be a blocker at this time, but the data is available if we want to get it, will just require some work with the events data structure

Reply:
Going to write up a ticket for a fast follow on this one.